packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -366,17 +366,13 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
                 // Update the contracts internal accounting of the amount of native asset in L2.
                 _balance -= _tx.value;
 
-                // Read the balance of the target contract before the transfer so the consistency
-                // of the transfer can be checked afterwards.
-                uint256 startBalance = IERC20(token).balanceOf(address(this));
-
                 // Transfer the ERC20 balance to the target, accounting for non standard ERC20
                 // implementations that may not return a boolean. This reverts if the low level
                 // call is not successful.
                 IERC20(token).safeTransfer({ to: _tx.target, value: _tx.value });
 
                 // The balance must be transferred exactly.
-                if (IERC20(token).balanceOf(address(this)) != startBalance - _tx.value) {
+                if (IERC20(token).balanceOf(address(this)) != _balance) {
```

Comment:
This is not safe, anybody can `ERC20.transfer` to this contract out of band and then brick deposits

Reply:
totally. maybe we should use `if (IERC20(token).balanceOf(address(this)) < _balance) { revert(); }`

Reply:
also, how is this different from the previous approach? looks like the contract could be bricked without any of the changes in this PR

Reply:
If you can produce a proof of concept that bricks the existing implementation, that would be amazing :)